pkg/reconciler/kubernetes/tektonchain/tektonchain.go

```diff
@@ -678,46 +676,19 @@ func AddControllerEnv(controllerEnvs []corev1.EnvVar) mf.Transformer {
 	}
 }
 
-func GenerateSigningSecrets(ctx context.Context) map[string][]byte {
+func generateSigningSecrets(ctx context.Context) map[string][]byte {
 	logger := logging.FromContext(ctx)
-	// Generate ECDSA key pair
-	privateKey, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
-	if err != nil {
-		logger.Error("Error generating private key:", err)
-		return nil
-	}
 
-	// Convert private key to PKCS8
-	privateKeyBytes, err := x509.MarshalPKCS8PrivateKey(privateKey)
+	// Here, nil indicates that no password is provided for the private key encryption, as it requires a user-provided secret.
+	keys, err := cosign.GenerateKeyPair(nil)
 	if err != nil {
-		logger.Error("Error marshaling private key:", err)
+		logger.Error("Error generating cosign key pair:", err)
 		return nil
 	}
 
-	// Encode private key to PEM
-	privateKeyPEM := pem.EncodeToMemory(&pem.Block{
-		Type:  "PRIVATE KEY",
-		Bytes: privateKeyBytes,
-	})
-
-	// Get public key
-	publicKey := &privateKey.PublicKey
-
-	// Marshal public key
-	publicKeyBytes, err := x509.MarshalPKIXPublicKey(publicKey)
-	if err != nil {
-		logger.Error("Error marshaling public key:", err)
-		return nil
-	}
-
-	// Encode public key to PEM
-	publicKeyPEM := pem.EncodeToMemory(&pem.Block{
-		Type:  "PUBLIC KEY",
-		Bytes: publicKeyBytes,
-	})
-
+	// Return the keys using cosign naming conventions.
 	return map[string][]byte{
-		"x509.pem":     privateKeyPEM,
-		"x509-pub.pem": publicKeyPEM,
+		"cosign.key": keys.PrivateBytes,
+		"cosign.pub": keys.PublicBytes,
```

Comment:
We currently use the `cosign generate-key-pair k8s://openshift-pipelines/signing-secrets` command to generate the key.
In addition to the `cosign.key` and `cosign.pub` we have a `cosign.password`, which I don't see in this code change.

Reply:
generating a random password make sens, it is better than have nothing :), @mbpavan we need to store in cosign.password as @Roming22 is mentionning

Reply:
Okay, makes sense. Added.